### PR TITLE
(#1124) Mobile Menu Spacing Bug

### LINKED ISSUE #1124: Fix white space between close button and first item when root menu first item does not have a child
The first item in the root menu should always have the same amount of white space between the close button and the start of the navigation regardless if the item has children or not.

Root menu first item of mobile menu without children
Screenshot 2023-07-18 at 2 52 28 PM

Root menu first item of mobile menu with children
Screenshot 2023-07-18 at 2 54 47 PM

We built it so when it's not a root menu, there would not be unnecessary padding when the Back button is present 
Screenshot 2023-07-18 at 3 20 58 PM

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #1124

### CODE REVIEW

**Review by andyvanavery31 — APPROVED** [review]
Passes Product Review.

File: packages/ncids-css/packages/nci-header/internal/_navigation-mobile.scss [hunk 1]
```diff
@@ -59,18 +59,21 @@
...
 			margin-top: units(8);
 		}
 	}
 }
 
+.nci-header-mobilenav > .nci-header-mobilenav__list > .nci-header-mobilenav__list-node:first-child:not(.active) {
+	border-top: 1px solid color("gray-warm-10");
+	margin-top: units(6);
+}
+.nci-header-mobilenav > .nci-header-mobilenav__list > .nci-header-mobilenav__list-item:first-child:not(.active) {
```
Comment (dlescarbeau) [inline packages/ncids-css/packages/nci-header/internal/_navigation-mobile.scss:73]: i believe this needs to be `.nci-header-mobilenav__nav > .nci-header-mobilenav__list` per `mobile-menu.ts` generation of the mobile menu

**Review by dlescarbeau — CHANGES_REQUESTED** [review]

File: packages/ncids-css/packages/nci-header/internal/_navigation-mobile.scss (hunk 1 above)
Comment (karlikpj) [reply packages/ncids-css/packages/nci-header/internal/_navigation-mobile.scss]: updating css and fixing storybook HTML to match

**Review by dlescarbeau — APPROVED** [review]

File: packages/ncids-css/packages/nci-header/internal/_navigation-mobile.scss [hunk 2]
```diff
@@ -56,21 +56,22 @@
...
 	}
 }
 
+.nci-header-mobilenav__nav > .nci-header-mobilenav__list > .nci-header-mobilenav__list-node:first-child:not(.active) {
+	border-top: 1px solid color("gray-warm-10");
+	margin-top: units(6);
+}
+.nci-header-mobilenav__nav > .nci-header-mobilenav__list > .nci-header-mobilenav__list-item:first-child:not(.active) {
+	border-top: 1px solid color("gray-warm-10");
+	margin-top: 3rem;
```
[the remark this answers is not in the document]
Comment (karlikpj) [reply packages/ncids-css/packages/nci-header/internal/_navigation-mobile.scss]: updated and building

**Review by arcepaul — APPROVED** [review]

**Pull request merged**